app/controllers/course/levels_controller.rb

```diff
@@ -6,32 +6,21 @@ class Course::LevelsController < Course::ComponentController
   def index #:nodoc:
   end
 
-  def new #:nodoc:
-  end
-
   def create #:nodoc:
-    if @level.save
-      redirect_to course_levels_path(current_course),
-                  success: t('.success', threshold: @level.experience_points_threshold)
-    else
-      render 'new'
-    end
-  end
+    # Delete all levels except the default level.
+    Course::Level.where(course_id: current_course.id).where.not(experience_points_threshold: 0).
```

Comment:
We should shift the updating of levels this into a method the levels model instead, rather than do the updating here in the controller, since it concerns with the implementation of levels.

Reply:
The Levels model is more for defining a single Level object though. Would it be better to put it under the Course model instead? Something like `current_course.set_levels(<array of exp thresholds>)`.

Reply:
It could be a class method under `Course::Levels`?